mitiq/pec/representations/learning.py

```diff
@@ -93,27 +98,107 @@ def learn_depolarizing_noise_parameter(
     )
 
     ideal_values = np.array(
-        [ideal_executor.evaluate(t, observable) for t in training_circuits]
+        [ideal_executor.evaluate(t, observable)[0] for t in training_circuits]
+    )
+
+    pec_data, method, minimize_kwargs = _parse_learning_kwargs(
+        learning_kwargs=learning_kwargs
     )
 
-    def depolarizing_noise_loss_function(
-        epsilon: List[int],
-        operations_to_mitigate: List[QPROGRAM],
-        training_circuits: List[QPROGRAM],
-        ideal_values: npt.NDArray[np.float64],
-        noisy_executor: Executor,
-        pec_kwargs: Dict["str", Any],
-        observable: Optional[Observable] = None,
-    ) -> float:
-        return biased_noise_loss_function(
-            np.array([epsilon[0], 0]),
-            operations_to_mitigate,
+    result = minimize(
+        biased_noise_loss_function,
+        [epsilon0, eta0],
+        args=(
+            operations_to_learn,
             training_circuits,
             ideal_values,
             noisy_executor,
             pec_kwargs,
+            pec_data,
             observable,
-        )
+        ),
+        method=method,
+        **minimize_kwargs,
+    )
+
+    success = result.success
+    epsilon_opt, eta_opt = result.x
+
+    return success, epsilon_opt, eta_opt
+
+
+def learn_depolarizing_noise_parameter(
+    operations_to_learn: List[QPROGRAM],
+    circuit: QPROGRAM,
+    ideal_executor: Executor,
+    noisy_executor: Executor,
+    pec_kwargs: Optional[Dict["str", Any]] = {},
+    num_training_circuits: int = 5,
+    fraction_non_clifford: float = 0.2,
+    training_random_state: Optional[np.random.RandomState] = None,
+    epsilon0: float = 0.05,
+    observable: Optional[Observable] = None,
+    **learning_kwargs: Dict["str", Any],
+) -> Tuple[bool, float]:
```

Comment:
`pec_kwargs` is important because it is the "knob" of the tradeoff between statistical error and execution time, which becomes all the more important in the optimization loop. The user will soon find they need to tune it for their specific problem.

Reply:
Gotcha, thanks for the explanation. Looks good then!